lib/xdrgen/generators/rust/src/types.rs

```diff
@@ -3,6 +3,9 @@ use core::{array::TryFromSliceError, fmt, fmt::Debug, marker::Sized, ops::Deref,
 #[cfg(feature = "std")]
 use core::marker::PhantomData;
 
+#[cfg(feature = "std")]
+const MAX_PREALLOCATED_BYTES_READ: usize = 1024; // 1KB
```

Comment:
Would be nice to make this configurable. But would rather set it to a sensible limit at this point and revisit that when it matters.